### PR TITLE
Fix Widget Kit Refreshing Bug 

### PR DESCRIPTION
**Pull request by manu-r12**
#### Note:
The problem was that RestAPIServices wasn’t refreshing properly when changing to new region due to some reasons. I added a method, getRefreshedAPIService, to fix this issue efficiently FOR NOW :) 

##### Preview:

https://github.com/user-attachments/assets/e541da68-8594-41d9-b1d4-e56f068709ae

### CODE REVIEW

File: OBAKitCore/Orchestration/CoreApplication.swift [hunk 1]
```diff
@@ -129,6 +129,15 @@ open class CoreApplication: NSObject,
         self.apiService = RESTAPIService(APIServiceConfiguration(baseURL: region.OBABaseURL, apiKey: config.apiKey, uuid: userUUID, appVersion: config.appVersion, regionIdentifier: region.regionIdentifier))
     }
 
+    public func getNewRefreshedRESTAPIService() -> RESTAPIService? {
```
Comment (aaronbrethorst) [inline OBAKitCore/Orchestration/CoreApplication.swift:132]: delete this

File: OBAWidget/Provider/WidgetDataProvider.swift [hunk 2]
```diff
@@ -10,48 +10,66 @@ import CoreLocation
...
+    )
+
     /// Loads arrivals and departures for all favorited bookmarks for the widget.
-    public func loadData() async {
-        guard let apiService = app.apiService else { return }
-        
+    func loadData() async {
+        arrDepDic = [:] 
+
+        guard let apiService = app.getNewRefreshedRESTAPIService() else {
```
Comment (aaronbrethorst) [inline OBAWidget/Provider/WidgetDataProvider.swift:53]: call `app.refreshServices()` and then use `app.apiService`

**Review by aaronbrethorst — CHANGES_REQUESTED** [review]
looks good! There's one change I'd like to see about how a new apiService object is constructed, but otherwise 🚀 .

Also, I'm not sure why the CI is failing. Tests are passing, so 🤷 . I'll get this merged once your change is in and we can worry about CI later.

File: OBAWidget/Provider/WidgetDataProvider.swift (hunk 2 above)
Comment (manu-r12) [reply OBAWidget/Provider/WidgetDataProvider.swift]: Done ✅

**Review by aaronbrethorst — APPROVED** [review]
great work 🚀

**Pull request merged**